src/leader.py

```diff
@@ -152,6 +157,35 @@ def lead(self):
             self.turn(code, position)
 
 
+    def demo_circle(self):
+        """
+        Controls the leader vehicle in a circle for demo purposes.
+
+        Sets the wheels and speed of the leader vehicle to drive in a circle.
+        """
+        self.turn_left()
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+        time.sleep(DEMO_CIRCLE_TIME)
```

Comment:
I agree, I originally envisioned it being a sort of listens for button, then enters a while loop so you could use timers instead of sleep, then exits the while loop when you press the button again, or maybe the a button puts the vehicle into demo mode, x and y are demo functions and a to cancel any or all demo functions.

Reply:
This is my understanding of it. Let's say that:

X -> demo_circle
Y -> demo_semicircles

The user presses the X button. The leader is now doing circles until any other button is pressed. If the button that was pressed was Y, stop doing circles and start doing semi-circles. If the user taps the right trigger, the leader vehicle should get out of the semi_circle mode and go back to waiting for input.

I figured that this functionality would be implemented when mapping the functions and that your functions would just be called repeatedly for as long as the system is in that mode.